zerver/tests/test_scheduled_messages.py

```diff
@@ -118,34 +119,27 @@ def test_successful_deliver_stream_scheduled_message(self) -> None:
         more_than_scheduled_delivery_datetime = (
             scheduled_message.scheduled_timestamp + datetime.timedelta(minutes=1)
         )
-        with mock.patch(
-            "zerver.actions.scheduled_messages.timezone_now",
-            return_value=more_than_scheduled_delivery_datetime,
-        ):
-            # mock time that will be set for delivered_message
-            with mock.patch(
-                "zerver.actions.message_send.timezone_now",
-                return_value=more_than_scheduled_delivery_datetime,
-            ):
-                result = try_deliver_one_scheduled_message(logger)
-                self.assertTrue(result)
-                logger.info.assert_called_once_with(
-                    "Sending scheduled message %s with date %s (sender: %s)",
-                    scheduled_message.id,
-                    scheduled_message.scheduled_timestamp,
-                    scheduled_message.sender_id,
-                )
-                scheduled_message.refresh_from_db()
-                assert isinstance(scheduled_message.delivered_message_id, int)
-                self.assertEqual(scheduled_message.delivered, True)
-                self.assertEqual(scheduled_message.failed, False)
-                delivered_message = Message.objects.get(id=scheduled_message.delivered_message_id)
-                self.assertEqual(delivered_message.content, scheduled_message.content)
-                self.assertEqual(
-                    delivered_message.rendered_content, scheduled_message.rendered_content
-                )
-                self.assertEqual(delivered_message.topic_name(), scheduled_message.topic_name())
-                self.assertEqual(delivered_message.date_sent, more_than_scheduled_delivery_datetime)
+        with time_machine.travel(more_than_scheduled_delivery_datetime):
```

Comment:
[Setting `tick=False`](https://github.com/adamchainz/time-machine#traveldestination--ticktrue) should fix your microsecond question, and make this test not potentially flaky.  In short, the default `.travel()` sets the time, but _time keeps moving forward_.  Since it takes a small but non-zero amount of time between when we enter this block (setting time to `more_than_scheduled_delivery_datetime`) and when the message is actually sent, the times disagree.  Telling the `time_travel` library that we want to go to a specific time _and stay there_ fixes the issue.

I also say that setting `tick=False` will make this not potentially flaky because `delivered_message.date_sent.replace(microsecond=0)` only works if the time didn't just roll over to a new second.  We know the starting `more_than_scheduled_delivery_datetime` is defined to be at the start of a second, but if the test was extremely slow for some reason (e.g. running massively parallel on a host, or virtualized where it didn't get scheduled for a second) then it's possible that it could take more than a second to get here.  Very unlikely, but it's not the code's fault if so!